File: core/src/main/java/com/pmcl/core/market/ModProject.java
package com.pmcl.core.market;

/**
 * 模组市场中的项目（CurseForge / Modrinth 通用模型）。
 */
public final class ModProject {

    private String source;       // "curseforge" | "modrinth"
    private String id;           // 项目 id（CurseForge 为数字字符串，Modrinth 为 slug/ID）
    private String slug;
    private String name;
    private String summary;
    private String author;
    private long downloadCount;
    private String iconUrl;
    private String websiteUrl;

    public ModProject(String source, String id, String slug, String name, String summary,
                      String author, long downloadCount, String iconUrl, String websiteUrl) {
        this.source = source;
        this.id = id;
        this.slug = slug;
        this.name = name;
        this.summary = summary;
        this.author = author;
        this.downloadCount = downloadCount;
        this.iconUrl = iconUrl;
        this.websiteUrl = websiteUrl;
    }

    public String getSource() { return source; }
    public String getId() { return id; }
    public String getSlug() { return slug; }
    public String getName() { return name; }
    public String getSummary() { return summary; }
    public String getAuthor() { return author; }
    public long getDownloadCount() { return downloadCount; }
    public String getIconUrl() { return iconUrl; }
    public String getWebsiteUrl() { return websiteUrl; }
}

File: pcmlAndroid/core/src/main/java/com/lash/pmcl/core/market/ModProject.kt
package com.lash.pmcl.core.market

/**
 * Mod 项目信息（搜索结果）。
 *
 * Android 版本：从 Java 移植，纯数据类，无平台依赖。
 */
data class ModProject(
    val source: String,
    val id: String,
    val slug: String,
    val name: String,
    val summary: String,
    val author: String,
    val downloadCount: Long,
    val iconUrl: String,
    val websiteUrl: String
)
